### ISSUE TITLE
Unicode characters are converted to `â€`

### ISSUE THREAD

**Issue by drinkitup777** [pgaskin/kepubify#76]
Platform: Windows 10 x64
kepubify version: v4.0.2 release from github

Symptoms: Unicode characters are converted to `â€`
Expected: Unicode characters are untouched

Steps to reproduce:
1. Insert the attached HTML into an epub (tested working with the [Gutenberg Frankenstein](https://www.gutenberg.org/ebooks/84))
2. Convert the resultant epub with `kepubify "bug_demo.epub"`
3. Observe that the last lines of the demo HTML should be endashes, but are converted to `â€`

Extra:
This is a strange bug. Deleting or inserting characters (of any value) seem to activate/de-activate it. I encountered this bug in a commercial epub (the text of which I've replaced with `x`s). I can give you more sample HTMLs if you need it.

---

Demo HTML to insert into test epub:
```html
<?xml version="1.0" encoding="utf-8" standalone="no"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN"
  "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
<body class="calibre">
  <p class="calibre22">xxxxxx xxx, xxx xx xxxxxx, xxxxxxxx, xxxx xxx xxxx.</p>

  <p class="calibre22">xxxx x xxxx xxxx xx xx xxxxxxxxx x xxxxxx xxx x xxxxxxxx xx xxx. xx xxx, xx xxxxxx, x xxxxxxxxx xxxxxxxxx: xxx xxxxx xxxxxxxxx xx xxxxxx xxxxx xxxx xxxxx. xxxxx xxxx xxx xxxxxxxx xxxxxxxxx xxxxxxxxx xx xxx xxxxxxxxx xxxxx xxxxxxxxx, xxx xxxxx xxx xx xxx xxxxxxxxx xxxxx xxxxxxxx xxxxxxxxx.</p>

  <p class="calibre22">xxxxxx xxxxxxxxxx xxxx xxx xxxxxxx xxxxx xxx xxxx xx xxxxx xxxxx xxxxx, xxx xxxxxxx xxx xxxxxxx xxxx xxx xxxxx xxxxxxx: xxx xxxxxxx; xxx xxxxxxxxx xxxxxxxxxx xxxxxxxxx xxxxxx; xxx xxxxx xxxxxxxx xxxxxx xxxxxxx xx xxx xxxx; xxxxxxx xxxxxxx; xxxxxxxx xxxxxxxxxxxxx; xxx xxxxxxx xxx xxx xxxxxxxx xxxxxx xxxx xxx xx xx xxx xxxxx xxxxxxx xxxx xxxxx.</p>

  <p class="calibre22">xxxxx xxx xx xxxx xxx.</p>
  
  <p class="calibre22">——————————————————————————————</p>
  <p class="calibre22">——————————————————————————————</p>
  <p class="calibre22">——————————————————————————————</p>
  <p class="calibre22">——————————————————————————————</p>
  <p class="calibre22">——————————————————————————————</p>
</body>
</html>
```

**Comment by pgaskin** [comment]
This would be a text encoding issue, probably caused by a regression in bd369afba303d121ab5e07b8778b74f2c607f71c due to [`golang.org/x/net/html/charset`](https://cs.opensource.google/go/x/net/+/fe4d6282:html/charset/charset.go) not reading the encoding in the XML declaration. Does this happen in kepubify v4.0.1?

**Comment by drinkitup777** [comment]
No, it doesn't. Version 4.0.1 works just fine. The new 4.0.2 is the one with the bug.

**Comment by pgaskin** [comment]
Ok, then it's what I suspected regarding the XML charset not being read. I'll fix that and release a new version in the next day or two.

**Comment by pgaskin** [comment]
Should be fixed in [`v4.0.3`](https://github.com/pgaskin/kepubify/releases/v4.0.3).

Thanks for the detailed bug report!

**Comment by drinkitup777** [comment]
Yup, fix confirmed on my end. Thanks for the quick response 😊
